packages/flutter/lib/src/material/tooltip.dart

```diff
@@ -114,11 +144,100 @@ class Tooltip extends StatefulWidget {
     this.child,
     this.triggerMode,
     this.enableFeedback,
-  }) : assert(message != null),
+  }) : assert(
+         message != null,
+         'A non-null String must be provided in message to a Tooltip widget.',
+       ),
+       tooltip = null,
+       semanticsLabel = null,
        super(key: key);
 
-  /// The text to display in the tooltip.
-  final String message;
+  /// Creates a tooltip using a widget provided to the [tooltip].
+  ///
+  /// **NOTE :** Using [Tooltip] and [Tooltip.custom], like examples given
+  /// below, will show the same results unless the custom [textStyle] or
+  /// [TooltipTheme] is not procvided for the [Tooltip]. Therefore, it is
+  /// recommended to use the [Tooltip] instead of [Tooltip.custom] in general
+  /// scenarios where just simple text tooltips needed like following examples.
+  /// ```dart
+  /// Tooltip(
+  ///   message:"my tooltip",
+  /// )
+  ///
+  /// Tooltip.custom(
+  ///   tooltip: Text("my tooltip"),
+  /// )
+  /// ```
+  ///
+  /// **NOTE :**
+  /// By default, tooltips should adhere to the
+  /// [Material specification](https://material.io/design/components/tooltips.html#spec).
+  /// But, in the case of [Tooltip.custom], this will ignore the [TooltipTheme.of] as widget
+  /// provided in the [tooltip] will use themes like as it would be using in the
+  /// normal application.
+  ///
+  /// All parameters that are defined in the constructor will
+  /// override the default values _and_ the values in [TooltipTheme.of].
+  const Tooltip.custom({
+    Key? key,
+    required Widget this.tooltip,
+    this.height,
+    this.padding,
+    this.margin,
+    this.verticalOffset,
+    this.preferBelow,
+    this.semanticsLabel,
+    this.decoration,
+    this.waitDuration,
+    this.showDuration,
+    this.child,
+    this.triggerMode,
+    this.enableFeedback,
+  }) : assert(
+         tooltip != null,
+         'A non-null Widget must be provided in tooltip to a Tooltip.custom widget.',
+       ),
+       message = null,
```

Comment:
I understand it but I just made it optional because if someone just wants to exclude the semantics, they will not provide the semanticsLabel otherwise it will be excluded. That is why I made it nullable like the Text.Rich constructor as I got the inspiration to introduce a new constructor for the Tooltip as well.

Reply:
Yes, but this is a completely different method from the other constructor for excluding semantics, and having two different ways to do it is confusing and unnecessary.